src/storage.rs

```diff
@@ -68,26 +68,49 @@ impl Default for AccountBasic {
     }
 }
 
-/// EVM Code, currently mapping to REVM's [ByteCode::LegacyAnalyzed].
-// TODO: Support raw legacy & EOF
 #[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
-pub struct EvmCode {
-    /// Bytecode with 32 zero bytes padding
+pub struct LegacyAnalyzedCode {
+    /// Bytecode with 32 zero bytes padding.
     bytecode: Bytes,
-    /// Original bytes length
+    /// Original bytes length.
     original_len: usize,
     /// Jump table.
     jump_table: Arc<BitVec<u8>>,
 }
 
+#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
+pub struct Eip7702Code {
+    /// EOA which which will inherit the bytecode.
+    delegated_address: Address,
+    /// Version of the bytecode.
+    version: u8,
+    /// Raw bytecode encoding MAGIC, version and delegated address.
+    raw: Bytes,
+}
+
+/// EVM Code, currently mapping to REVM's [ByteCode].
+// TODO: Support raw legacy & EOF
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum EvmCode {
+    /// Analyzed legacy code, with jump table.
+    LegacyAnalyzed(LegacyAnalyzedCode),
+    /// Delegated code.
+    Eip7702(Eip7702Code),
+}
+
 impl From<EvmCode> for Bytecode {
     fn from(code: EvmCode) -> Self {
         // TODO: Better error handling.
         // A common trap would be converting a default [EvmCode] into
         // a [Bytecode]. On failure we should fallback to legacy and
         // analyse again.
-        unsafe {
-            Bytecode::new_analyzed(code.bytecode, code.original_len, JumpTable(code.jump_table))
+        match code {
+            EvmCode::Eip7702(code) => {
+                Bytecode::Eip7702(Eip7702Bytecode::new(code.delegated_address))
```

Comment:
Shouldn't we construct a full `Bytecode::Eip7702` by "forwarding" all the fields here? This conversion seems to create an empty code regardless of the current `code`.